core/embed/trezorhal/drbg.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __TREZORHAL_DRBG_H__
+#define __TREZORHAL_DRBG_H__
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "assert.h"
+#include "chacha_drbg.h"
+#include "entropy.h"
+
+#define DRBG_INIT_NONCE_LENGTH 0
+#define DRBG_INIT_TRNG_ENTROPY_LENGTH 50
+_Static_assert(CHACHA_DRBG_DERIVATION_FUNCTION_BLOCK_LENGTH -
+                       CHACHA_DRBG_DERIVATION_FUNCTION_PREFIX_LENGTH -
+                       CHACHA_DRBG_DERIVATION_FUNCTION_PADDING ==
+                   DRBG_INIT_TRNG_ENTROPY_LENGTH + DRBG_INIT_NONCE_LENGTH,
+               "");
+// Make sure entropy in chacha_drbg derivation function fills exactly one block
+// of hashing function. This is not needed it's just an optimalization.
+
+#define DRBG_MIX_HW_ENTROPY_TRNG_ENTROPY_LENGTH 6
```

Comment:
So the DRBG has 48 bits of randomness to start with? If we generated a seed from this before the device had time to properly reseed, then that would totally defeat security measures like long PIN or SD protect.

Reply:
Actually, it looks like this does break long PIN and SD protect, because we generate the storage data encryption key upon startup, which almost certainly happens before the DRBG has time to properly reseed.

Reply:
The drbg is initialised by `drbg_init` with 50 * 8 bits of entropy.